### ISSUE TITLE
Absolutely confused... WAN 2.2 Lora

### ISSUE THREAD

**Issue by RandomKaos999** [kohya-ss/musubi-tuner#462]
Hi all,

I've trained a lora with Musubi Tuner for WAN 2.2 14B T2V.

With the samples, the character is damn well perfect - no issues there at all.
I converted the chosen Lora using the convert lora python script provided, using "other" and tried to use it within ComfyUI. Videos generated with comfy.. no where near, absolutely no where near the correct likeness.

This is part of the batch file I used to train:

```
set vae=G:\GenAI\Models\WanTraining\wan_2.1_vae.safetensors
set t5=G:\GenAI\Models\WanTraining\models_t5_umt5-xxl-enc-bf16.pth
set dit_low=G:\GenAI\Models\unet\Video\wan2.2_t2v_low_noise_14B_fp16.safetensors
set dit_high=G:\GenAI\Models\unet\Video\wan2.2_t2v_high_noise_14B_fp16.safetensors

set command=accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 "G:\GenAI\Software-New\musubi-tuner\src\musubi_tuner\wan_train_network.py" --task t2v-A14B --dit %dit_low%  --dataset_config "config\dataset.toml" --sdpa --mixed_precision fp16 --optimizer_type adamw8bit --learning_rate %learning_rate% --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module networks.lora_wan --network_dim 32 --network_alpha 32 --timestep_sampling sigmoid --discrete_flow_shift 12.0 --max_train_epochs %max_train_epochs% --save_every_n_epochs 1 --seed 42 --output_dir orig_models --output_name %output_name% --logging_dir=logs --save_state --save_every_n_epochs 1 --save_last_n_epochs_state 5 --offload_inactive_dit --fp8_base --dit_high_noise %dit_high%
```

Based on the above, it produced a single Lora file... which I have assumed needs to be added to ComfyUI using the Load Lora Model Only node, on both the low noise AND the high noise models. Ive ensured that they dont cross paths either as well. (If im wrong on this, please tell me).

That's basically the jist of how im generating with ComfyUI, nothing special other than the template provided by ComfyUI and adding the Load Lora Model Only nodes and updating the prompt.

So since I couldn't get a viable output via ComfyUI, i decided to try the original loras produced via MT using an inference batch file:

```
set lora_path="WAN22_512512_HIGHANDLOW_0001-000100.safetensors"
set trigger=(name here)

set vae=G:\GenAI\Models\WanTraining\wan_2.1_vae.safetensors
set t5=G:\GenAI\Models\WanTraining\models_t5_umt5-xxl-enc-bf16.pth
set dit_low=G:\GenAI\Models\unet\Video\wan2.2_t2v_low_noise_14B_fp16.safetensors
set dit_high=G:\GenAI\Models\unet\Video\wan2.2_t2v_high_noise_14B_fp16.safetensors

python "G:\GenAI\Software-New\musubi-tuner\src\musubi_tuner\wan_generate_video.py" --fp8 --task t2v-A14B --video_size  720 1280 --video_length 33 --infer_steps 20 --prompt "%trigger%. %trigger% is standing in a bar. She has voluminous hair styled in tight curls. She is directly facing the viewer." --save_path "output_test_video\save.mp4" --output_type both --dit %dit_low% --vae %vae% --dit_high_noise %dit_high% --t5 %t5% --attn_mode torch --lazy_loading --lora_weight %lora_path% --lora_weight_high_noise %lora_path%
```

The output from this was absolutely near enough perfect, likeness was at 95%. So im thinking im either converting them wrong, or im somehow using the lora wrong in ComfyUI.

Can someone please point me in the right direction? Feel free to copy and paste the issue and just say "duh" with a bit of info for me. I can take it lol. It might be something massively trivial that i've overlooked.

I don't get any errors in the console with ComfyUI either, in regards to anything not matching or any other hints. It just generates, but there's no likeness at all.

**Comment by kohya-ss** [comment]
I haven't used ComfyUI much, but I think your training method and inference method in ComfyUI are correct.

The LoRA conversion may not be necessary. Could you try loading it in ComfyUI without converting it?

**Comment by obsxrver** [comment]
~~ComfyUI works best with a separate high-noise and low-noise model for WAN 2.2.~~ 

Some people recommend producing two LoRAs by training the model on the same dataset on both the high noise and low noise models, using different timesteps. This is mentioned in the [docs](https://github.com/kohya-ss/musubi-tuner/blob/main/docs/wan.md). This is how I do it, and what the guide I follow recommends. 

Following this guide has gotten me spectacular results on everything I tried so far. It also works for concepts, haven't tried a style, but it probably does that well, too. best loras I have ever trained by a mile. 
https://civitai.com/articles/17740

For those of you living in censorious nanny-states, I made a paste of the guide [(here)](https://github.com/obsxrver/wan22-lora-training/)

**Comment by RandomKaos999** [comment]
Thanks for the responses.

After testing conversion ISN'T required for the lora's to work, however comparing the samples to the actual loras within ComfyUI, it seems that your inference with MT produces likeness samples much much earlier based on the script above (0.0002 LR). I had to wait till about 6500 steps on 2 different datasets for the likeness to be accurate on ComfyUI, but samples from your MT inference script was producing likness around 1800 steps or so for each.

@obsxrver - sadly civitai is blocked for my country, so I cant even check any guides or anything for ideas.

Can I suggest updating the documentation so it doesnt state it needs to be converted for use with ComfyUI in case there's anyone else being baffled with the same thing - it states to convert to other, or at least thats how I read it.

**Comment by obsxrver** [comment]
Edited for clarity. Hope it helps!

**Comment by jarrodthebobo** [comment]
Not to bump this old issue but Im experiencing a similar issue with a recent character lora I've been attempting to train; during the training process the sample videos that are output look very similar to the character, but once loading into comfyui the similarity drops quite a lot and I have to crank the strength of the high and low lora up near 2 for the similarity to creep back in... but by that point things start to look pretty crunchy. 

This is without other loras loaded and attempting to use multiple different WAN models to ensure its not model specific. I've attempted converting the lora weights as well but cant see much of a difference. Im at a loss here.
